ivy/functional/ivy/experimental/norms.py

```diff
@@ -195,20 +195,28 @@ def batch_norm(
     inv = 1.0 / ivy.sqrt(variance + eps)
     if scale is not None:
         inv = inv * scale
-    xnormalized = x * inv.astype(x.dtype, copy=False) + ivy.astype(
-        offset - mean * inv if offset is not None else -mean * inv, x.dtype
+    xnormalized = x * inv.astype(ivy.dtype(x), copy=False) + ivy.astype(
```

Comment:
ig given that `x` is guaranteed to be an `ivy.Array` we could continue to use `x.dtype` right?